rocAL/source/readers/file_source_reader.cpp

```diff
@@ -60,30 +83,38 @@ Reader::Status FileSourceReader::initialize(ReaderConfig desc) {
     _loop = desc.loop();
     _meta_data_reader = desc.meta_data_reader();
     _last_batch_info = desc.get_last_batch_policy();
+    _pad_last_batch_repeated = _last_batch_info.second;
+    _stick_to_shard = desc.get_stick_to_shard();
+    _shard_size = desc.get_shard_size();
     ret = subfolder_reading();
+    _curr_file_idx = get_start_idx();  // shard's start_idx would vary for every shard in the vector
```

Comment:
start and for each shard should be stored in a list as precomputed

Reply:
changes done